go-controller/pkg/ovn/address_set_syncer/address_set_sync_test.go

```diff
@@ -0,0 +1,700 @@
+				Policies: []string{"lrp1"},
+			},
+		}
+		testSyncerWithData(testData, initialDb, nil, controllerName, false)
+	})
+	ginkgo.It("preserves unknown ExternalIDs", func() {
+		initialAS := createInitialAS(hybridRoutePolicyPrefix+"node1_v4", []string{testIPv4})
+		initialAS.ExternalIDs["unknown_id"] = "id_value"
+		testData := []asSync{
+			{
+				before:                initialAS,
+				after:                 syncerToBuildData.getHybridRouteAddrSetDbIDs("node1"),
+				addressSetFactoryIPID: ipv4AddressSetFactoryID,
+				afterTweak: func(as *nbdb.AddressSet) {
+					as.ExternalIDs["unknown_id"] = "id_value"
+				},
+			},
+		}
+		testSyncerWithData(testData, nil, nil, controllerName, false)
+	})
+	// verify different address set owners
+	ginkgo.It("updates address set owned by HybridNodeRouteOwnerType and its references", func() {
+		asName := hybridRoutePolicyPrefix + "node1_v4"
+		hashedASName := hashedAddressSet(asName)
+		testData := []asSync{
+			{
+				before:                createInitialAS(asName, []string{testIPv4}),
+				after:                 syncerToBuildData.getHybridRouteAddrSetDbIDs("node1"),
+				addressSetFactoryIPID: ipv4AddressSetFactoryID,
+			},
+		}
+		initialDb := []libovsdbtest.TestData{
+			&nbdb.LogicalRouterPolicy{
+				UUID:     "lrp1",
+				Priority: types.HybridOverlayReroutePriority,
+				Action:   nbdb.LogicalRouterPolicyActionReroute,
+				Nexthops: []string{testGatewayIPv4},
+				Match:    fmt.Sprintf(`inport == "%s%s" && ip4.src == $%s`, types.RouterToSwitchPrefix, "node1", hashedASName),
+			},
+			&nbdb.LogicalRouter{
+				UUID:     types.OVNClusterRouter + "-UUID",
+				Name:     types.OVNClusterRouter,
+				Policies: []string{"lrp1"},
+			},
+		}
+		testSyncerWithData(testData, initialDb, nil, controllerName, false)
+	})
+	ginkgo.It("updates address set owned by EgressQoSOwnerType and its references", func() {
+		asName := egressQoSRulePrefix + "namespace-123_v4"
+		hashedASName := hashedAddressSet(asName)
+		testData := []asSync{
+			{
+				before:                createInitialAS(asName, []string{testIPv4}),
+				after:                 syncerToBuildData.getEgressQosAddrSetDbIDs("namespace", "123"),
+				addressSetFactoryIPID: ipv4AddressSetFactoryID,
+			},
+		}
+		initialDb := []libovsdbtest.TestData{
+			&nbdb.QoS{
+				Direction:   nbdb.QoSDirectionToLport,
+				Match:       "(ip4.dst == 1.2.3.4/32) && ip4.src == $" + hashedASName,
+				Priority:    qosPriority,
+				Action:      map[string]int{nbdb.QoSActionDSCP: 40},
+				ExternalIDs: map[string]string{"EgressQoS": "namespace"},
+				UUID:        "qos1-UUID",
+			},
+		}
+		testSyncerWithData(testData, initialDb, nil, controllerName, false)
+	})
+	ginkgo.It("updates address set owned by EgressQoSOwnerType and its references, check batching", func() {
+		// this test checks the batching works for number of address sets = 5, which is 2.5 test batch sizes.
+		// it also checks disabled batching.
+		for _, disableBatching := range []bool{false, true} {
```

Comment:
nit: it is better to have distinct test cases for batching vs no batching. Or at least distinct test case failure messages. To be able to distinguish which one of the two options failed within the test case.

Reply:
pushed this change